pumpkin/src/error.rs

```diff
@@ -65,3 +67,39 @@ impl PumpkinError for DeserializerError {
         None
     }
 }
+
+impl PumpkinError for WorldError {
+    fn is_kick(&self) -> bool {
+        false
+    }
+
+    fn severity(&self) -> log::Level {
+        log::Level::Warn
+    }
+
+    fn client_kick_reason(&self) -> Option<String> {
+        None
+    }
+}
+
+impl PumpkinError for PlayerError {
+    fn is_kick(&self) -> bool {
+        match self {
+            Self::BlockOutOfReach => false,
+            Self::InvalidBlockFace => true,
+        }
+    }
+
+    fn severity(&self) -> log::Level {
+        match self {
+            Self::BlockOutOfReach | Self::InvalidBlockFace => log::Level::Warn,
+        }
+    }
+
+    fn client_kick_reason(&self) -> Option<String> {
+        match self {
+            Self::BlockOutOfReach => None,
+            Self::InvalidBlockFace => Some("Invalid block face".into()),
+        }
+    }
+}
```

Comment:
No. Don't implement the PumpkinError trait here. Do it below the structs

Reply:
InventoryError's impl was here, should I move that too?

Reply:
And DeserializerError as well, all PumpkinError impls were here actually